### PR TITLE
Improvements to the Gradle build

### PR DESCRIPTION
**Pull request by bsiegel**

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -0,0 +1,2 @@
+include ':sdk:core:azure-core'
+include ':sdk:storage:azure-storage-common'
```
Comment (anuchandy) [inline settings.gradle]: any advantage in moving `settings.xml` file from `sdk` dir? With this here we have to point AndroidStudio to the root, root contains other artifacts not used for daily dev, wondering whether it affects editor experience.

File: sdk/core/azure-core/build.gradle [hunk 2]
```diff
@@ -35,13 +43,10 @@ android {
 dependencies {
     implementation fileTree(dir: "libs", include: ["*.jar"])
     implementation "androidx.annotation:annotation:$annotationsVersion"
+    implementation "com.fasterxml.jackson.core:jackson-databind:$jacksonDatabindVersion"
+    implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-xml:$jacksonDataFormatXmlVersion"
     implementation "com.jakewharton.threetenabp:threetenabp:$threeTenAbpVersion"
     implementation "com.squareup.okhttp3:okhttp:$okHttpVersion"
     implementation "com.squareup.retrofit2:retrofit:$retrofitVersion"
-    implementation "com.jakewharton.threetenabp:threetenabp:$threeTenAbpVersion"
```
Comment (vcolin7) [inline sdk/core/azure-core/build.gradle]: We still need the this dependency for the AddDateInterceptor

Comment (anuchandy) [reply sdk/core/azure-core/build.gradle]: seems this was a duplicate entry, there is same in line48

File: settings.gradle (hunk 1 above)
Comment (anuchandy) [reply settings.gradle]: I cloned this PR and opened in AndroidStudio. Though it showed everything when opened first time, after the load/sync there exists only the two modules (core and storage-common). We're good here.

**Pull request merged**